File: frontend/src/lib/tools.ts
/**
 * Agent/工具配置中心（前端镜像）
 *
 * 与后端 src-tauri/src/tools.rs 保持同步，字段含义相同：
 *   id         — 命令行 --agent 标识符，也是数据库 tool 字段的值
 *   name       — 显示名称
 *   projectDir — 项目级 Skills 目录（相对于项目根目录）
 *   globalDir  — 全局 Skills 目录（相对于 $HOME）
 *   color      — UI 配色（Hex）
 */
export interface ToolConfig {
  id: string
  name: string
  projectDir: string
  globalDir: string
  color: string
}

export const ALL_TOOLS: ToolConfig[] = [
  { id: 'amp',            name: 'Amp',             projectDir: '.agents/skills',      globalDir: '.config/agents/skills',       color: '#6366F1' },
  { id: 'antigravity',    name: 'Antigravity',     projectDir: '.agent/skills',       globalDir: '.gemini/antigravity/skills',  color: '#8B5CF6' },
  { id: 'augment',        name: 'Augment',         projectDir: '.augment/skills',     globalDir: '.augment/skills',             color: '#06B6D4' },
  { id: 'claude-code',    name: 'Claude Code',     projectDir: '.claude/skills',      globalDir: '.claude/skills',              color: '#D97706' },
  { id: 'cline',          name: 'Cline',           projectDir: '.cline/skills',       globalDir: '.cline/skills',               color: '#7C3AED' },
  { id: 'codebuddy',      name: 'CodeBuddy',       projectDir: '.codebuddy/skills',   globalDir: '.codebuddy/skills',           color: '#0284C7' },
  { id: 'codex',          name: 'Codex',           projectDir: '.agents/skills',      globalDir: '.codex/skills',               color: '#059669' },
  { id: 'command-code',   name: 'Command Code',    projectDir: '.commandcode/skills', globalDir: '.commandcode/skills',         color: '#B45309' },
  { id: 'continue',       name: 'Continue',        projectDir: '.continue/skills',    globalDir: '.continue/skills',            color: '#0891B2' },
  { id: 'cortex',         name: 'Cortex Code',     projectDir: '.cortex/skills',      globalDir: '.snowflake/cortex/skills',    color: '#29B5E8' },
  { id: 'crush',          name: 'Crush',           projectDir: '.crush/skills',       globalDir: '.config/crush/skills',        color: '#DB2777' },
  { id: 'cursor',         name: 'Cursor',          projectDir: '.agents/skills',      globalDir: '.cursor/skills',              color: '#8B5CF6' },
  { id: 'droid',          name: 'Droid',           projectDir: '.factory/skills',     globalDir: '.factory/skills',             color: '#475569' },
  { id: 'gemini-cli',     name: 'Gemini CLI',      projectDir: '.agents/skills',      globalDir: '.gemini/skills',              color: '#4285F4' },
  { id: 'github-copilot', name: 'GitHub Copilot',  projectDir: '.agents/skills',      globalDir: '.copilot/skills',             color: '#24292F' },
  { id: 'goose',          name: 'Goose',           projectDir: '.goose/skills',       globalDir: '.config/goose/skills',        color: '#2563EB' },
  { id: 'iflow-cli',      name: 'iFlow CLI',       projectDir: '.iflow/skills',       globalDir: '.iflow/skills',               color: '#7C3AED' },
  { id: 'junie',          name: 'Junie',           projectDir: '.junie/skills',       globalDir: '.junie/skills',               color: '#E11D48' },
  { id: 'kilo',           name: 'Kilo Code',       projectDir: '.kilocode/skills',    globalDir: '.kilocode/skills',            color: '#0E7490' },
  { id: 'kimi-cli',       name: 'Kimi Code CLI',   projectDir: '.agents/skills',      globalDir: '.config/agents/skills',       color: '#2563EB' },
  { id: 'kiro-cli',       name: 'Kiro CLI',        projectDir: '.kiro/skills',        globalDir: '.kiro/skills',                color: '#0284C7' },
  { id: 'kode',           name: 'Kode',            projectDir: '.kode/skills',        globalDir: '.kode/skills',                color: '#9333EA' },
  { id: 'mcpjam',         name: 'MCPJam',          projectDir: '.mcpjam/skills',      globalDir: '.mcpjam/skills',              color: '#64748B' },
  { id: 'mistral-vibe',   name: 'Mistral Vibe',    projectDir: '.vibe/skills',        globalDir: '.vibe/skills',                color: '#F97316' },
  { id: 'mux',            name: 'Mux',             projectDir: '.mux/skills',         globalDir: '.mux/skills',                 color: '#6B7280' },
  { id: 'openclaw',       name: 'OpenClaw',        projectDir: 'skills',              globalDir: '.openclaw/skills',            color: '#DC2626' },
  { id: 'opencode',       name: 'OpenCode',        projectDir: '.agents/skills',      globalDir: '.config/opencode/skills',     color: '#16A34A' },
  { id: 'openhands',      name: 'OpenHands',       projectDir: '.openhands/skills',   globalDir: '.openhands/skills',           color: '#7C3AED' },
  { id: 'pi',             name: 'Pi',              projectDir: '.pi/skills',          globalDir: '.pi/agent/skills',            color: '#0F766E' },
  { id: 'qoder',          name: 'Qoder',           projectDir: '.qoder/skills',       globalDir: '.qoder/skills',               color: '#0369A1' },
  { id: 'qwen-code',      name: 'Qwen Code',       projectDir: '.qwen/skills',        globalDir: '.qwen/skills',                color: '#7C2D12' },
  { id: 'replit',         name: 'Replit',          projectDir: '.agents/skills',      globalDir: '.config/agents/skills',       color: '#F97316' },
  { id: 'roo',            name: 'Roo Code',        projectDir: '.roo/skills',         globalDir: '.roo/skills',                 color: '#4F46E5' },
  { id: 'trae',           name: 'Trae',            projectDir: '.trae/skills',        globalDir: '.trae/skills',                color: '#EC4899' },
  { id: 'trae-cn',        name: 'Trae CN',         projectDir: '.trae/skills',        globalDir: '.trae-cn/skills',             color: '#F43F5E' },
  { id: 'universal',      name: 'Universal',       projectDir: '.agents/skills',      globalDir: '.config/agents/skills',       color: '#6B7280' },
  { id: 'windsurf',       name: 'Windsurf',        projectDir: '.windsurf/skills',    globalDir: '.codeium/windsurf/skills',    color: '#0EA5E9' },
  { id: 'zencoder',       name: 'Zencoder',        projectDir: '.zencoder/skills',    globalDir: '.zencoder/skills',            color: '#7C3AED' },
  { id: 'neovate',        name: 'Neovate',         projectDir: '.neovate/skills',     globalDir: '.neovate/skills',             color: '#16A34A' },
  { id: 'pochi',          name: 'Pochi',           projectDir: '.pochi/skills',       globalDir: '.pochi/skills',               color: '#E11D48' },
  { id: 'adal',           name: 'AdaL',            projectDir: '.adal/skills',        globalDir: '.adal/skills',                color: '#0284C7' },
]

/** 快速查找 Map */
export const toolMap = new Map(ALL_TOOLS.map((t) => [t.id, t]))

/** tool id → 显示名称（向后兼容 utils.ts 的 toolNames） */
export const toolNames: Record<string, string> = Object.fromEntries(
  ALL_TOOLS.map((t) => [t.id, t.name])
)

/** tool id → 配色（向后兼容 utils.ts 的 toolColors） */
export const toolColors: Record<string, string> = Object.fromEntries(
  ALL_TOOLS.map((t) => [t.id, t.color])
)

/** 所有工具 ID 组成的列表（用于下拉选择等） */
export const toolIds = ALL_TOOLS.map((t) => t.id)

File: frontend/src-tauri/src/tools.rs
/// Agent/工具配置中心
///
/// 每条记录描述一个支持 Skills 的 AI 编程工具：
///   - `id`          : 命令行 --agent 标识符，也是数据库 tool 字段的值
///   - `name`        : 显示名称
///   - `project_dir` : 项目级 Skills 目录（相对于项目根目录）
///   - `global_dir`  : 全局 Skills 目录（相对于 $HOME）
#[derive(Debug, Clone, Copy)]
pub struct ToolConfig {
    pub id: &'static str,
    pub name: &'static str,
    pub project_dir: &'static str,
    pub global_dir: &'static str,
}

pub const ALL_TOOLS: &[ToolConfig] = &[
    ToolConfig { id: "amp",           name: "Amp",             project_dir: ".agents/skills",      global_dir: ".config/agents/skills"        },
    ToolConfig { id: "antigravity",   name: "Antigravity",     project_dir: ".agent/skills",       global_dir: ".gemini/antigravity/skills"   },
    ToolConfig { id: "augment",       name: "Augment",         project_dir: ".augment/skills",     global_dir: ".augment/skills"              },
    ToolConfig { id: "claude-code",   name: "Claude Code",     project_dir: ".claude/skills",      global_dir: ".claude/skills"               },
    ToolConfig { id: "cline",         name: "Cline",           project_dir: ".cline/skills",       global_dir: ".cline/skills"                },
    ToolConfig { id: "codebuddy",     name: "CodeBuddy",       project_dir: ".codebuddy/skills",   global_dir: ".codebuddy/skills"            },
    ToolConfig { id: "codex",         name: "Codex",           project_dir: ".agents/skills",      global_dir: ".codex/skills"                },
    ToolConfig { id: "command-code",  name: "Command Code",    project_dir: ".commandcode/skills", global_dir: ".commandcode/skills"          },
    ToolConfig { id: "continue",      name: "Continue",        project_dir: ".continue/skills",    global_dir: ".continue/skills"             },
    ToolConfig { id: "cortex",        name: "Cortex Code",     project_dir: ".cortex/skills",      global_dir: ".snowflake/cortex/skills"     },
    ToolConfig { id: "crush",         name: "Crush",           project_dir: ".crush/skills",       global_dir: ".config/crush/skills"         },
    ToolConfig { id: "cursor",        name: "Cursor",          project_dir: ".agents/skills",      global_dir: ".cursor/skills"               },
    ToolConfig { id: "droid",         name: "Droid",           project_dir: ".factory/skills",     global_dir: ".factory/skills"              },
    ToolConfig { id: "gemini-cli",    name: "Gemini CLI",      project_dir: ".agents/skills",      global_dir: ".gemini/skills"               },
    ToolConfig { id: "github-copilot",name: "GitHub Copilot",  project_dir: ".agents/skills",      global_dir: ".copilot/skills"              },
    ToolConfig { id: "goose",         name: "Goose",           project_dir: ".goose/skills",       global_dir: ".config/goose/skills"         },
    ToolConfig { id: "iflow-cli",     name: "iFlow CLI",       project_dir: ".iflow/skills",       global_dir: ".iflow/skills"                },
    ToolConfig { id: "junie",         name: "Junie",           project_dir: ".junie/skills",       global_dir: ".junie/skills"                },
    ToolConfig { id: "kilo",          name: "Kilo Code",       project_dir: ".kilocode/skills",    global_dir: ".kilocode/skills"             },
    ToolConfig { id: "kimi-cli",      name: "Kimi Code CLI",   project_dir: ".agents/skills",      global_dir: ".config/agents/skills"        },
    ToolConfig { id: "kiro-cli",      name: "Kiro CLI",        project_dir: ".kiro/skills",        global_dir: ".kiro/skills"                 },
    ToolConfig { id: "kode",          name: "Kode",            project_dir: ".kode/skills",        global_dir: ".kode/skills"                 },
    ToolConfig { id: "mcpjam",        name: "MCPJam",          project_dir: ".mcpjam/skills",      global_dir: ".mcpjam/skills"               },
    ToolConfig { id: "mistral-vibe",  name: "Mistral Vibe",    project_dir: ".vibe/skills",        global_dir: ".vibe/skills"                 },
    ToolConfig { id: "mux",           name: "Mux",             project_dir: ".mux/skills",         global_dir: ".mux/skills"                  },
    ToolConfig { id: "openclaw",      name: "OpenClaw",        project_dir: "skills",              global_dir: ".openclaw/skills"             },
    ToolConfig { id: "opencode",      name: "OpenCode",        project_dir: ".agents/skills",      global_dir: ".config/opencode/skills"      },
    ToolConfig { id: "openhands",     name: "OpenHands",       project_dir: ".openhands/skills",   global_dir: ".openhands/skills"            },
    ToolConfig { id: "pi",            name: "Pi",              project_dir: ".pi/skills",          global_dir: ".pi/agent/skills"             },
    ToolConfig { id: "qoder",         name: "Qoder",           project_dir: ".qoder/skills",       global_dir: ".qoder/skills"                },
    ToolConfig { id: "qwen-code",     name: "Qwen Code",       project_dir: ".qwen/skills",        global_dir: ".qwen/skills"                 },
    ToolConfig { id: "replit",        name: "Replit",          project_dir: ".agents/skills",      global_dir: ".config/agents/skills"        },
    ToolConfig { id: "roo",           name: "Roo Code",        project_dir: ".roo/skills",         global_dir: ".roo/skills"                  },
    ToolConfig { id: "trae",          name: "Trae",            project_dir: ".trae/skills",        global_dir: ".trae/skills"                 },
    ToolConfig { id: "trae-cn",       name: "Trae CN",         project_dir: ".trae/skills",        global_dir: ".trae-cn/skills"              },
    ToolConfig { id: "universal",     name: "Universal",       project_dir: ".agents/skills",      global_dir: ".config/agents/skills"        },
    ToolConfig { id: "windsurf",      name: "Windsurf",        project_dir: ".windsurf/skills",    global_dir: ".codeium/windsurf/skills"     },
    ToolConfig { id: "zencoder",      name: "Zencoder",        project_dir: ".zencoder/skills",    global_dir: ".zencoder/skills"             },
    ToolConfig { id: "neovate",       name: "Neovate",         project_dir: ".neovate/skills",     global_dir: ".neovate/skills"              },
    ToolConfig { id: "pochi",         name: "Pochi",           project_dir: ".pochi/skills",       global_dir: ".pochi/skills"                },
    ToolConfig { id: "adal",          name: "AdaL",            project_dir: ".adal/skills",        global_dir: ".adal/skills"                 },
];

/// 根据工具 ID 获取配置
pub fn get_tool(id: &str) -> Option<&'static ToolConfig> {
    ALL_TOOLS.iter().find(|t| t.id == id)
}

/// 获取所有不重复的项目级目录（用于 watcher 监听和 reconcile 扫描）
pub fn unique_project_dirs() -> Vec<&'static str> {
    let mut dirs: Vec<&'static str> = ALL_TOOLS.iter().map(|t| t.project_dir).collect();
    dirs.dedup_by(|a, b| a == b);
    dirs.sort_unstable();
    dirs.dedup();
    dirs
}
